### PR TITLE
Normalise all paths to be relative to plz root when running gofmt

### PR DESCRIPTION
**Pull request by marcuscaisey**
This PR rewrites `Service.normalizePaths`  so that absolute paths are transformed to be relative to the plz root. I've added some tests which cover these edge cases. 

This change fixes a couple of bugs:

#### Replicating
```
#!/usr/bin/env bash

mkdir -p root/parent/child
cd root

plz init --no_prompt >/dev/null

cat << EOF > parent/child/main.go
package main

import "fmt"

func main() {
    fmt.Println("foo")
}
EOF

cd parent
wollemi gofmt $PWD/child

find . -name BUILD.plz
```

##### Expected Output
```
INFO[0000] modified                                      path=/parent/child
./child/BUILD.plz
```

#### Output
```

```

#### Replicating
```
#!/usr/bin/env bash

mkdir -p root/child
cd root

plz init --no_prompt >/dev/null

cat << EOF > child/main.go
package main

import "github.com/example/foo"

func main() {
    foo.Foo()
}
EOF

cat << EOF > child/.wollemi.json
{
    "known_dependency": {
        "github.com/example/foo": "//third_party/go/foo"
    }
}
EOF

wollemi gofmt $PWD/child

find . -name BUILD.plz
```

##### Expected output
```
INFO[0000] modified                                      path=/child
./child/BUILD.plz
```

##### Output
```
ERRO[0000] could not resolve go import                   go_import=github.com/example/foo path=/Users/marcus/scratch/wollemi-absolute-path-bug/root/child process=create rule=child
```

### CODE REVIEW

File: domain/wollemi/service.go [hunk 1]
```diff
@@ -62,14 +62,16 @@ func (this *Service) normalizePaths(paths []string) []string {
...
-	if this.wd != this.root && strings.HasPrefix(this.wd, this.root) {
-		relpath = strings.TrimPrefix(this.wd, this.root+"/")
-
-		for i, path := range paths {
-			paths[i] = filepath.Join(relpath, strings.TrimSuffix(path, "/"))
+	for i, path := range paths {
+		if !filepath.IsAbs(path) {
+			path = filepath.Join(this.wd, path)
+		}
+		// TODO: should we check that all absolute paths live under the root before we get to here? i don't think we can
```
Comment (brian-bice) [inline domain/wollemi/service.go:69]: I agree @marcuscaisey. Attempting to format a path outside of the root does not make sense. Could you add a check for that?

Comment (brian-bice) [reply domain/wollemi/service.go:69]: Sorry for the delay. Thanks for the fix @marcuscaisey. This MR looks great to me.

File: third_party/go/github.com/golang/BUILD.plz [hunk 2]
```diff
@@ -1,9 +1,11 @@
 package(default_visibility = ["PUBLIC"])
 
+GOMOCK_VERSION = "1.6.0"
```
Comment (marcuscaisey) [inline third_party/go/github.com/golang/BUILD.plz:3]: upgraded this so that i could run tests on m1 mac

**Comment by marcuscaisey** [comment]
updated with absolute path validation @brian-bice, sorry it took so long to get around to this

**Comment by brian-bice** [comment]
Thanks @marcuscaisey. Looks like there are some merge conflicts. The `third_party/go` deps were recently rearranged and the `github.com/golang` package was moved to `github.com/golang/mock`.

**Comment by marcuscaisey** [comment]
> Thanks @marcuscaisey. Looks like there are some merge conflicts. The `third_party/go` deps were recently rearranged and the `github.com/golang` package was moved to `github.com/golang/mock`.

fixed. whilst i've got you here, i was thinking of adding a field to the config to change the default build file name like 
```
{
  "build_file_name": "BUILD"
}
```
since we always use `BUILD` as our build file name at work. wdyt? (i would keep `BUILD.plz` as the default for backwards compatibility)

**Comment by brian-bice** [comment]
@marcuscaisey that's something I've actually been meaning to address for awhile but unfortunately I just haven't had time. The new config value would be okay. However, the `.plzconfig` already defines a `BuildFileName` so I think a better solution would be for `wollemi` to use the exiting configuration from the `.plzconfig` instead.

Screen Shot 2022-04-15 at 11 35 03 AM

In theory it should be pretty easy to add. That change would definitely be welcome if you are up for doing it?

**Comment by marcuscaisey** [comment]
> @marcuscaisey that's something I've actually been meaning to address for awhile but unfortunately I just haven't had time. The new config value would be okay. However, the `.plzconfig` already defines a `BuildFileName` so I think a better solution would be for `wollemi` to use the exiting configuration from the `.plzconfig` instead.
> 
> Screen Shot 2022-04-15 at 11 35 03 AM
> 
> In theory it should be pretty easy to add. That change would definitely be welcome if you are up for doing it?

yeah taking it from the config is definitely cleaner, good shout. and yeah i'm happy to take this on

**Comment by brian-bice** [comment]
Great. Thanks again @marcuscaisey.

**Comment by marcuscaisey** [comment]
no worries 👍
